tests/components/xiaomi_miio/test_select.py

```diff
@@ -0,0 +1,267 @@
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    CONF_MAC,
+    DOMAIN as XIAOMI_DOMAIN,
+    KEY_DEVICE,
+)
+from homeassistant.components.xiaomi_miio.select import (
+    AttributeEnumMapping,
+    XiaomiMiioSelectDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_HOST,
+    CONF_MODEL,
+    CONF_TOKEN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+from tests.components.xiaomi_miio import TEST_MAC
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DummyEnum_0(enum.Enum):
+    """Enum to test auto naming."""
+
+    Option_0 = 0
+    Option_1 = 1
+
+
+class DummyEnum_1(enum.Enum):
+    """Enum to test predefined naming."""
+
+    BadName_0 = 0
+    BadName_1 = 1
+
+
+class DummySelectStatus:
+    """Dummy device status."""
+
+    def __init__(self):
+        """Initialize the dummy device status."""
+        self.dummy = DummyEnum_0.Option_1
+        self.bad_name = DummyEnum_1.BadName_1
+
+
+class DummySelectDevice:
+    """Dummy device."""
+
+    def __init__(self, model, *args, **kwargs):
+        """Initialize the dummy device."""
+        self.data = DummySelectStatus()
+
+    def status(self):
+        """Retrieve properties."""
+        return self.data
+
+    def set_dummy_value(self, value: DummyEnum_0):
+        """Set dummy enum value."""
+        self.data.dummy = value
+        return True
+
+    def set_bad_name_value(self, value: DummyEnum_1):
+        """Set bad name enum value."""
+        self.data.bad_name = value
+        return True
+
+
+TEST_MODEL_TO_ATTR_MAP: dict[str, list] = {
+    "Dummy": [
+        AttributeEnumMapping("dummy", DummyEnum_0),
+        AttributeEnumMapping("bad_name", DummyEnum_1),
+    ]
+}
+
+TEST_SELECTOR_TYPES = (
+    XiaomiMiioSelectDescription(
+        key="dummy",
+        attr_name="dummy",
+        name="Dummy",
+        set_method="set_dummy_value",
+        set_method_error_message="Set dummy value failed.",
+        options=("option_0", "option_1"),
+    ),
+    XiaomiMiioSelectDescription(
+        key="bad_name",
+        attr_name="bad_name",
+        name="Bad Name",
+        options_map={
+            "BadName_0": "New_0",
+            "BadName_1": "New_1",
+        },
+        set_method="set_bad_name_value",
+        set_method_error_message="Set bad name value failed.",
+        options=("new_0", "new_1"),
+    ),
+)
+
+
+async def mock_async_create_miio_device_and_coordinator(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> None:
+    """Set up a data coordinator and one miio device to service multiple entities."""
+    name = entry.title
+    update_method = _async_update_data_default
+
+    device = DummySelectDevice("Dummy")
+
+    # Create update miio device and coordinator
+    coordinator = DataUpdateCoordinator(
```

Comment:
We don't need to create a coordinator as that's created by the integration when we set up the integration by setting up a MockConfigEntry.